src/coreclr/nativeaot/Runtime/arm64/WriteBarriers.asm

```diff
@@ -328,17 +330,17 @@ ExchangeRetry
 ;;   x15  : the object reference (RHS of the assignment).
 ;;
 ;; On exit:
-;;   x9   : trashed
-;;   x12  : trashed if WRITE_BARRIER_CHECK
-;;   x14  : trashed
+;;   x12  : trashed
+;;   x15  : trashed
+;;   x17  : trashed if WRITE_BARRIER_CHECK
 ;;
     LEAF_ENTRY RhpAssignRefArm64, _TEXT
     ALTERNATE_ENTRY RhpAssignRefAVLocation
     ALTERNATE_ENTRY RhpAssignRefX1AVLocation
 
         stlr    x15, [x14]
 
-        INSERT_UNCHECKED_WRITE_BARRIER_CORE x14, x15, x14
+        INSERT_UNCHECKED_WRITE_BARRIER_CORE x14, x15, x15
```

Comment:
Trash `x15` instead of `x14` for consistency with the two other barriers, `RhpCheckedAssignRefArm64` and 
`RhpByRefAssignRefArm64`